src/order_cmd.cpp

```diff
@@ -1961,34 +1961,37 @@ bool UpdateOrderDest(Vehicle *v, const Order *order, int conditional_depth, bool
 			}
 
 			if (v->current_order.GetDepotActionType() & ODATFB_NEAREST_DEPOT) {
-				/* We need to search for the nearest depot (hangar). */
-				ClosestDepot closestDepot = v->FindClosestDepot();
-
-				if (closestDepot.found) {
-					/* PBS reservations cannot reverse */
-					if (pbs_look_ahead && closestDepot.reverse) return false;
-
-					v->SetDestTile(closestDepot.location);
-					v->current_order.SetDestination(closestDepot.destination);
-
-					/* If there is no depot in front, reverse automatically (trains only) */
-					if (v->type == VEH_TRAIN && closestDepot.reverse) Command<CMD_REVERSE_TRAIN_DIRECTION>::Do(DC_EXEC, v->index, false);
-
-					if (v->type == VEH_AIRCRAFT) {
-						Aircraft *a = Aircraft::From(v);
-						if (a->state == FLYING && a->targetairport != closestDepot.destination) {
-							/* The aircraft is now heading for a different hangar than the next in the orders */
-							AircraftNextAirportPos_and_Order(a);
+				/* Try to delay the search for a bit if the previous destination is invalid. */
```

Comment:
A possible answer to @PeterN's request. 🙂 

```suggestion
				/* If the vehicle can't find its destination, delay its next search.
				 * In case many vehicles are in this state, use the vehicle index to spread out pathfinder calls.*/
```